R-package/R/create-release-files.R

```diff
@@ -0,0 +1,411 @@
+source(file.path(getwd(), "R", "odm-dictionary-file.R"))
+source(file.path(getwd(), "R", "files.R"))
+#' Create release files
+#'
+#' Creates release files given the user OSF link and auth token.
+#'
+#' @param OSF_LINK_RELEASE link to the dictionary stored on OSF used for updating.
+#' @param OSF_LINK_PAST_RELEASE link to the dictionary stored on OSF used for removal of previous release files.
+#' @param OSF_TOKEN OSF auth token used for modifying OSF directories
+#' @param dictionary_path optional string containing path to the dictionary directory. When provided no dictionary is downloaded.
+#' @param past_dictionary_path optional string containing path to the previous release dictionary directory. When provided no dictionary is downloaded.
+#'
+#' @export
+create_release_files <-
+  function(OSF_LINK_RELEASE,
+           OSF_LINK_PAST_RELEASE,
+           OSF_TOKEN,
+           dictionary_path = NULL,
+           past_dictionary_path = NULL) {
+    # Setup logging
+    # Remove previous log
+    file.remove(odm_dictionary$log_path)
+    # Set logger appending
+    logger::log_appender(logger::appender_file(odm_dictionary$log_path))
+    
+    # Download file using passed credentials
+    if (is.null(dictionary_path)) {
+      dictionary_path <- odm_dictionary$tmp_dictionary_directory
+      osfr::osf_auth(OSF_TOKEN)
+      osfr::osf_retrieve_file(OSF_LINK_RELEASE) %>%
+        osfr::osf_download(path = dictionary_path)
+    }
+    # Validate dictionary version
+    dictionary_info <- validate_version(dictionary_path)
+    
+    dictionary <- dictionary_info[[1]]
+    dictionary_version <- dictionary_info[[2]]
+    
+    # Validate files sheet
+    files_to_make <-
+      validate_files_sheet(dictionary,
+                           dictionary_version)
+    
+    create_files(files_to_make,
+                 dictionary)
+    
+    # Download previous release dictionary
+    if (is.null(past_dictionary_path)) {
+      past_dictionary_path <-
```

Comment:
This logic is the same as [downloading the release file](https://github.com/Big-Life-Lab/PHES-ODM/blob/file-removal/R-package/R/create-release-files.R#L27-L32). It should be pulled into a function.